### PR TITLE
boot: get rid of lib.uniq

### PR DESCRIPTION
**Pull request by Mic92**
Fixes https://github.com/numtide/nixos-facter-modules/issues/35

### CODE REVIEW

File: modules/nixos/boot.nix [hunk 1]
```diff
@@ -3,6 +3,8 @@ let
 
   cfg = config.facter.boot;
   inherit (config.facter) report;
+  collectDriver = list: lib.foldl' (lst: value: lst ++ value.driver_modules or [ ]) [ ] list;
```
Comment (Mic92) [inline modules/nixos/boot.nix:6]: Also, does it make a difference if we use `driver_modules` vs. `driver_module` here?

Comment (brianmcgee) [reply modules/nixos/boot.nix:6]: Yeah, they're distinct things.

Comment (brianmcgee) [inline modules/nixos/boot.nix:6]: To match the previous logic you need to target `driver_module` not `driver_modules`.

**Review by brianmcgee — CHANGES_REQUESTED** [review]

File: modules/nixos/boot.nix (hunk 1 above)
Comment (Mic92) [reply modules/nixos/boot.nix:6]: I intentionally used `driver_modules` here because of https://github.com/numtide/nixos-facter-modules/issues/35

Comment (Mic92) [reply modules/nixos/boot.nix:6]: https://git.sr.ht/~r-vdp/nixos-config/tree/main/item/hosts/beelink/facter.json#L685

Comment (brianmcgee) [reply modules/nixos/boot.nix:6]: On second thought, I guess we want to catch all the possible driver modules? I'm a little unclear between the distinction of `driver_module` and `driver_modules` if I'm being honest. 

I guess `driver_module` is the currently active one, with `driver_modules` being the possible applicable modules?

You can see this example that @r-vdp shared:

```json
"driver": "usb-storage",
"driver_module": "usb_storage",
"drivers": ["sd", "usb-storage"],
"driver_modules": ["sd_mod", "usb_storage"],
"module_alias": "usb:v0781p5583d0100dc00dsc00dp00ic08isc06ip50in00"
```

Comment (Mic92) [reply modules/nixos/boot.nix:6]: "driver_module" is the "main module". "driver_modules" are all modules needed.

Comment (brianmcgee) [reply modules/nixos/boot.nix:6]: For reference it is attempting to replicate the behaviour here:

https://github.com/NixOS/nixpkgs/blob/dac9cdf8c930c0af98a63cbfe8005546ba0125fb/nixos/modules/installer/tools/nixos-generate-config.pl#L122-L148

Comment (brianmcgee) [reply modules/nixos/boot.nix:6]: > "driver_module" is the "main module". "driver_modules" are all modules needed for this device.

In which case your logic makes more sense, and should address #35 at the same time as you can see in the example above `sd_mod` is included in the `driver_modules`.

**Review by brianmcgee — APPROVED** [review]

File: modules/nixos/boot.nix (hunk 1 above)
Comment (Mic92) [reply modules/nixos/boot.nix:6]: wrong paste. hang on.

Comment (Mic92) [reply modules/nixos/boot.nix:6]: ls -la /sys/bus/scsi/devices/
total 0
drwxr-xr-x 2 root root 0 Sep 24 16:37 .
drwxr-xr-x 4 root root 0 Sep 24 16:37 ..
lrwxrwxrwx 1 root root 0 Jan 21  2016 1:0:0:0 -> ../../../devices/platform/ff5c0000.usb/usb3/3-1/3-1.1/3-1.1:1.0/host1/target1:0:0/1:0:0:0
lrwxrwxrwx 1 root root 0 Jan 21  2016 2:0:0:0 -> ../../../devices/platform/ff5c0000.usb/usb3/3-1/3-1.4/3-1.4:1.0/host2/target2:0:0/2:0:0:0
lrwxrwxrwx 1 root root 0 Sep 11 13:19 host0 -> ../../../devices/platform/ff600000.usb/xhci-hcd.0.auto/usb5/5-1/5-1:1.0/host0
lrwxrwxrwx 1 root root 0 Jan 21  2016 host1 -> ../../../devices/platform/ff5c0000.usb/usb3/3-1/3-1.1/3-1.1:1.0/host1
lrwxrwxrwx 1 root root 0 Jan 21  2016 host2 -> ../../../devices/platform/ff5c0000.usb/usb3/3-1/3-1.4/3-1.4:1.0/host2
lrwxrwxrwx 1 root root 0 Sep 24 16:48 target1:0:0 -> ../../../devices/platform/ff5c0000.usb/usb3/3-1/3-1.1/3-1.1:1.0/host1/target1:0:0
lrwxrwxrwx 1 root root 0 Sep 24 16:48 target2:0:0 -> ../../../devices/platform/ff5c0000.usb/usb3/3-1/3-1.4/3-1.4:1.0/host2/target2:0:0

Comment (Mic92) [reply modules/nixos/boot.nix:6]: So using both drivers is indeed correct. `usb_storage` doesn't have any module dependency on `sd_mod`, so we have to add both.

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**